src/node/audio_buffer_source.rs

```diff
@@ -0,0 +1,673 @@
+use crossbeam_channel::{Receiver, Sender};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+use crate::audio_buffer::AudioBuffer;
+use crate::context::{AsBaseAudioContext, AudioContextRegistration, AudioParamId};
+use crate::param::{AudioParam, AudioParamOptions, AutomationRate};
+use crate::render::{AudioParamValues, AudioProcessor, AudioRenderQuantum};
+use crate::{AtomicF64, SampleRate, RENDER_QUANTUM_SIZE};
+
+use super::{AudioNode, ChannelConfig, ChannelConfigOptions};
+
+// attributes shared between the node and the renderer
+//
+// @note - kind of pack both Scheduler and Controller with an additionnal
+//  `duration` entry and `seek` renamed to `offset`
+// @note - this could probably be generated by a macro! (`Vec<{name, default<type>}>`)
```

Comment:
I'd rather avoid macro's unless really necessary. It's like magic and I think we should avoid magic!

Reply:
makes sens

Reply:
I think however that would be nice to have kind of `Atomic` parameter bag that could be defined per Node, with proper names and types. Don't really know how to achieve that with Rust (maybe another issue too?)